### ISSUE TITLE
Local File Support for InstallerURL Property in Manifest Files

### ISSUE THREAD

**Issue by excelsi** [microsoft/winget-cli#4361]
### Description of the new feature / enhancement

I described the Feature here as an Issue:
https://github.com/microsoft/winget-cli/issues/4358

Desired are local relative / designated Paths for the InstallerURL Property in Package Manifest File like:
./Microsoft Visual Studio Code_1.88.0_User_X64_inno_en-US.exe (relative to the same Directory like the manifest file)
and/or
D:/TEMP/tempdown/Microsoft Visual Studio Code_1.88.0_User_X64_inno_en-US.exe

If the Feature Request is ok and its no Issue you can close the Issue.

Thanks!

**Comment by excelsi** [comment]
Any news on this?
Thanks!

**Comment by denelon** [comment]
What scenario does this add value? Is this for "local" manifest validation, or something else where just running the local installer directly doesn't suffice?

**Comment by madcampos** [comment]
> What scenario does this add value? Is this for "local" manifest validation, or something else where just running the local installer directly doesn't suffice?

For me, there is no installer, it is a zip file and I would like to take advantage of winget creating a nice entry to uninstall the program and adding links to `PATH`.

**Comment by excelsi** [comment]
> What scenario does this add value? Is this for "local" manifest validation, or something else where just running the local installer directly doesn't suffice?

I have a local distribution Scenario without any internet access where a local Installer Path is needed.

**Comment by aserfass-msft** [comment]
In an offline scenario it would be nice to point to a local copy of the installer, validate the hash with winget, and install with winget (simplified commands). My team would also find this valuable. 

A prime example would be run: "winget download <packageName>" 
then
"winget install -m <yamlFileFromDownloadCommand>"
If the generated yaml file referenced the local/relative file path that would enable many offline install scenarios

**Comment by denelon** [comment]
We're looking into the scenario for installing "offline" store packages. We'll need to extend the schema to add the license file to the manifest, and at that point, we'll look at options for referencing a "local" installer.

**Comment by excelsi** [comment]
> We're looking into the scenario for installing "offline" store packages. We'll need to extend the schema to add the license file to the manifest, and at that point, we'll look at options for referencing a "local" installer.

Great thanks for the Feedback!

**Comment by excelsi** [comment]
@denelon will there be a preview of this feature in a preview releases still in this year or starting next year? Thanks!

**Comment by noyard** [comment]
> @denelon will there be a preview of this feature in a preview releases still in this year or starting next year? Thanks!

It depends upon what priority this item gets.  Right now there are only 4 thumbs up on your April 8th post (https://github.com/microsoft/winget-cli/issues/4361#issue-2231813640).  Hint: thumbs up your post ;)

**Comment by FaffeF** [comment]
For me, this feature would add more value if Winget would search for resources in some common path, or a cache that could be pre-populated via a download command

**Comment by excelsi** [comment]
@noyard now.i have 8 😉
I hope im not at at the end of a 1000 items wishlist 🤣
Thanks!

**Comment by pl4nty** [comment]
@denelon could pre-populating the winget cache be a supported scenario for non-store apps?

**Comment by excelsi** [comment]
@noyard @denelon  any chance to get the feature or a preview until the end of this year? Thanks!

**Comment by daveMueller** [comment]
> I have a local distribution Scenario without any internet access where a local Installer Path is needed. I also dont have a local Webserver in this case. But it still would be nice to use winget as Installer (with --manifest as local which works already).

I totally agree to this. There are plenty of applications that for instance are running in a regulated environment, hence they don't have internet access and hosting a local webserver isn't a really nice solution. 
For instance, I'm currently working for a company that deploys software on a dedicated device in operating rooms. Five years ago we decided for a more managed deployment on the `Win10IoT` host OS and choose to go with `Chocolatey` as the package manager (`WinGet` wasn't releast at that time).
`Chocolatey` offers the option to have local sources. We expected that this would also be possible with `WinGet` as for instance `APT` also has this feature. Now we will migrate our device to `Win11IoT` and also wanted to exchange `Chocolatey` with `WinGet` to remove the dependency to a third-party tool. But to my surprise this missing feature is blocking us from doing that.

**Comment by pl4nty** [comment]
@daveMueller I can share a workaround with you via email if you're interested, but it's probably unsupported. don't want to share it publicly until I've spoken with denelon

**Comment by excelsi** [comment]
> [@daveMueller](https://github.com/daveMueller) I can share a workaround with you via email if you're interested, but it's probably unsupported. don't want to share it publicly until I've spoken with denelon

@pl4nty Thanks, would be nice if you can share it with me too!

**Comment by Ivan171** [comment]
> [@daveMueller](https://github.com/daveMueller) I can share a workaround with you via email if you're interested, but it's probably unsupported. don't want to share it publicly until I've spoken with denelon

Can you share it with me as well?

**Comment by gitscholli** [comment]
Any news on this? This feature would come in handy in my scenarios, too.  :-)

**Comment by anthonyedwards587** [comment]
> [@daveMueller](https://github.com/daveMueller) I can share a workaround with you via email if you're interested, but it's probably unsupported. don't want to share it publicly until I've spoken with denelon

that would be very handy for us in our block all outbound type setup

**Comment by rolfste** [comment]
@daveMueller, can you share it with me as well ? Thanks !!!

**Comment by excelsi** [comment]
@noyard @denelon is there any rough timeplan when this feature can appear in a preview/beta version?
Thanks for your efforts!

**Comment by gst** [comment]
Hi there, any follow up here ?

@daveMueller can you share the "workaround" you have ? Thank you.

**Comment by noyard** [comment]
I wrote my own workaround  see https://github.com/noyard/CreateWingetPSADTPackages

Look in Invoke-AppDeployToolkit.ps1,   I copy the binary files to %temp%/winget folder using the hash of the file.  Then when winget goes to install... The files are already there    you just have to customize it for your specific use.

**Comment by daveMueller** [comment]
@gst we tried a bunch of workarounds but nothing felt stable enough to use it in a medical device. We then decided to just stay with `Chocolatey` because we already had all the packages and our pipelines set up for it.
